drivers/fpga/fpga_ice40.c

```diff
@@ -589,6 +614,21 @@ static int fpga_ice40_init(const struct device *dev)
 	BUILD_ASSERT(FPGA_ICE40_TRAILING_CLOCKS(inst) >= FPGA_ICE40_TRAILING_CLOCKS_MIN);          \
 	BUILD_ASSERT(FPGA_ICE40_TRAILING_CLOCKS(inst) <= UINT8_MAX);                               \
 	BUILD_ASSERT(FPGA_ICE40_MHZ_DELAY_COUNT(inst) >= 0);                                       \
+	BUILD_ASSERT(ICE40_FPGA_LOAD_MODE_GPIO_AVAILABLE ||                                        \
+		     !DT_INST_PROP(inst, load_mode_bitbang));                                      \
+	BUILD_ASSERT(ICE40_FPGA_LOAD_MODE_SPI_AVAILABLE || DT_INST_PROP(inst, load_mode_bitbang)); \
+	BUILD_ASSERT(!DT_INST_PROP(inst, load_mode_bitbang) ||                                     \
+		     DT_INST_NODE_HAS_PROP(inst, creset_gpios));                                   \
+	BUILD_ASSERT(!DT_INST_PROP(inst, load_mode_bitbang) ||                                     \
+		     DT_INST_NODE_HAS_PROP(inst, cdone_gpios));                                    \
+	BUILD_ASSERT(!DT_INST_PROP(inst, load_mode_bitbang) ||                                     \
+		     DT_INST_NODE_HAS_PROP(inst, clk_gpios));                                      \
+	BUILD_ASSERT(!DT_INST_PROP(inst, load_mode_bitbang) ||                                     \
+		     DT_INST_NODE_HAS_PROP(inst, pico_gpios));                                     \
+	BUILD_ASSERT(!DT_INST_PROP(inst, load_mode_bitbang) ||                                     \
+		     DT_INST_NODE_HAS_PROP(inst, gpios_set_reg));                                  \
+	BUILD_ASSERT(!DT_INST_PROP(inst, load_mode_bitbang) ||                                     \
+		     DT_INST_NODE_HAS_PROP(inst, gpios_clear_reg));                                \
```

Comment:
How about using [2 different compatibles](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/sensor/maxim/ds18b20/ds18b20.c#L359-L369) both targeting this same driver?

That way, it might be possible to move all these build time asserts into devicetree bindings, while also enforcing that only one of them is chosen for every given instance.